common/crypto/platform/openssl/crypto.cpp

```diff
@@ -147,11 +182,135 @@ ExpectedPrivateKey PrivateKey::LoadFromPEM(
 			"Failed to load the key: " + private_key_path + " " + GetOpenSSLErrorMessage()));
 	}
 
-	return unique_ptr<PrivateKey>(new PrivateKey(std::move(private_key)));
+	return PrivateKey(std::move(private_key));
 }
 
-ExpectedPrivateKey PrivateKey::LoadFromPEM(const string &private_key_path) {
-	return PrivateKey::LoadFromPEM(private_key_path, "");
+#ifdef MENDER_CRYPTO_OPENSSL_LEGACY
+ExpectedPrivateKey PrivateKey::LoadFromHSM(const Args &args) {
+	log::Trace("Loading the private key from HSM");
+
+	ENGINE_load_builtin_engines();
+	auto engine = EnginePtr(ENGINE_by_id(args.ssl_engine.c_str()), engine_free_func);
+
+	if (engine == nullptr) {
+		return expected::unexpected(MakeError(
+			SetupError,
+			"Failed to get the " + args.ssl_engine
+				+ " engine. No engine with the ID found: " + GetOpenSSLErrorMessage()));
+	}
+	log::Debug("Loaded the HSM engine successfully!");
+
+	int res = ENGINE_init(engine.get());
+	if (not res) {
+		return expected::unexpected(MakeError(
+			SetupError,
+			"Failed to initialise the hardware security module (HSM): "
+				+ GetOpenSSLErrorMessage()));
+	}
+	log::Debug("Successfully initialised the HSM engine");
+
+	auto private_key = unique_ptr<EVP_PKEY, void (*)(EVP_PKEY *)>(
+		ENGINE_load_private_key(
+			engine.get(),
+			args.private_key_path.c_str(),
+			(UI_METHOD *) NULL,
+			NULL /*callback_data */),
```

Comment:
Should use `nullptr`.

Reply:
I've seen that in other work we have used `NULL` when interacting with `C`, so I stuck to that here. 
I can change it, but just so we are clear

Reply:
Right. It's probably because we copied it from C examples or whatever. I'd use `nullptr` since it has an important type safety role.